django_informixdb_vault/base.py

```diff
@@ -168,9 +167,13 @@ def get_connection_params(self):
         # parse/get conn_params from django_informixdb
         conn_params = super().get_connection_params()
 
-        # We don't actually use USER and PASSWORD, so delete them
-        del self.settings_dict['USER']
-        del self.settings_dict['PASSWORD']
+        # We don't actually use USER and PASSWORD, so delete them.
+        # As this method can be called from multiple threads, it's
+        # possible that they have already been deleted.
+        if 'USER' in self.settings_dict:
+            del self.settings_dict['USER']
```

Comment:
What made you choose the `if` approach, rather than `try` or `lock` as you mentioned in #6 ?
Is there a chance of a race condition / mutation between the time the `if` is evaluated, and the `del` is executed?

Reply:
I initially tried wrapping just the dictionary deletions in a lock / semaphore, but when running tests, this didn't solve the problem, i.e. you would need to wrap a larger block of code with a lock / semaphore, and I was concerned that this would negate the value of multi-threading.

The try / except method might be a tiny bit more reliable than the if / else method - I'm open to switching to that if you prefer, but lately I've been questioning introducing exceptions in terms of making the code easily testable.  I guess there's not much difference in this case.  Given that there are 10 threads each time the test runs and it runs for 3 different Python versions, and I've run all tests several times locally, my feeling is that the if / else method is reliable.

Reply:
Switched to try/except method for improved thread safety

Reply:
Thanks!